portal/portal-api/api/src/java/org/sakaiproject/portal/api/Redirect.java

```diff
@@ -0,0 +1,26 @@
+package org.sakaiproject.portal.api;
+
+/**
+ * This is to store redirect url and appendPath information
+ * for redirect of a site in hierarchy and it's subsites
+ */
+public class Redirect {
+	private String url;
```

Comment:
Something like this is an ideal case to be imutable. We know everything about  it when we're creating it and we should never need to change it later on. Having a constructor that takes all the arguments and final member variables is a good way todo this.